### ISSUE TITLE
[HOLD for payment 2024-10-08][$250] Task – `.com` appears after assignee avatar when create a task manualy

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49454]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.38-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4983838
**Email or phone of affected tester (no customers):** ponikarchuks+118924@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Log in as a Gmail account
3. Open any chat
4. Enter [] @ and select Expensify user, wright task title and send
Example:
[] @[applausetester+jpcategory_2@applause.expensifail.com](mailto:applausetester+jpcategory_2@applause.expensifail.com)
New task

## Expected Result:
`.com` not appears after assignee avatar

## Actual Result:
`.com` appears after assignee avatar

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/cde0eeb8-e402-499c-8ba9-76b577211b18

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021836890784466753823</li>
        <li>Upwork Job ID: 1836890784466753823</li>
        <li>Last Price Increase: 2024-09-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Ollyws</details>

**Comment by IuliiaHerets** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by CyberAndrii** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Task – `.com` appears after assignee avatar when create a task manualy

### What is the root cause of that problem?

This regex does not handle subdomains in emails correctly.

https://github.com/Expensify/App/blob/37bf6b2fef128786dfcd83257a56f56c0cd1eb3d/src/pages/home/report/ReportFooter.tsx#L126-L132

### What changes do you think we should make in order to solve the problem?

Use this regex:

```js
const taskRegex = /^\[\]\s+(?:@([^\s@]+(?:\.\w+)+))?\s*([\s\S]*)/;

...

const title = match[3] ? match[3].trim().replace(/\n/g, ' ') : undefined;
```

**Comment by mallenexpensify** [comment]
Unable to repro on staging, Desktop.  @Ollyws can you attempt reproduction plz? 
<img width="402" alt="image" src="https://github.com/user-attachments/assets/f8c01b63-b947-499f-bbdb-9cdbcffd636f">

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by CyberAndrii** [comment]
@mallenexpensify you need to also enter the task title after an email. For example `New task`

<img width="481" alt="Screenshot 2024-09-20 at 01 42 08" src="https://github.com/user-attachments/assets/645a56cd-fb2a-44d8-a4fd-79a35c961f42">

**Comment by iwiznia** [comment]
Is the regex supposed to only match `[] user@email.com` and `[] @someuser` types of texts?
Asking because it also matches things like `[] some@thing` which I think is invalid and would break things?

**Comment by bernhardoj** [comment]
The email [regex from MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#basic_validation) matches `some@thing`. Maybe we can use that regex and make the domain optional.

```
^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+(@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*)?$
// simplified
^[\w.!#$%&'*+/=?^`{|}~-]+(@[a-zA-Z\d](?:[a-zA-Z\d-]{0,61}[a-zA-Z\d])?(?:\.[a-zA-Z\d](?:[a-zA-Z\d-]{0,61}[a-zA-Z\d])?)*)?$
```

**Comment by iwiznia** [comment]
> Maybe we can use that regex and make the domain optional.

That sounds good to me

**Comment by bernhardoj** [comment]
PR is ready

cc: @Ollyws

**Comment by mallenexpensify** [comment]
Sorry @NickTooker , wrong label :ohnothing:

**Comment by mallenexpensify** [comment]
Contributor: @bernhardoj due $250 via NewDot 
Contributor+: @Ollyws due $250 via NewDot  

@Ollyws plz complete the BZ checklist. 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:
- [ ]

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by garrettmknight** [comment]
$250 approved for @Ollyws

**Comment by mallenexpensify** [comment]
Test case created
- https://github.com/Expensify/Expensify/issues/434928

Thanks all!

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
